typegen/src/lib.rs

```diff
@@ -12,7 +12,21 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//! A library based on [scale-info](https://github.com/paritytech/scale-info) to transpile portable registries of types into rust type definitions.
+#![deny(unused_crate_dependencies)]
+#![deny(missing_docs)]
+
+// The #![deny(unused_crate_dependencies)] requires us to do these for the example to work:
+#[cfg(test)]
+use frame_metadata as _;
+#[cfg(test)]
+use prettyplease as _;
+#[cfg(test)]
+use scale_bits as _;
```

Comment:
@lexnv would `cargo machete` work better than that unused dependencies check (I saw it mentioned in Subxt but haven't dug into it at all yet :))